### ISSUE TITLE
Use of long int in Otter not compatible with igraph

### ISSUE THREAD

**Issue by adamtuft** [adamtuft/otter#7]
There appears to be a problem with igraph correctly reading long and unsigned long ints as vertex IDs:

Steps to reproduce:
`git clone git clone git@github.com:bsc-pm/bots.git`
`cd bots/`
`./configure && make`
`OMP_TOOL_LIBRARIES=/home/adam/git/otter/lib/libotter.so run/run-fib.sh -i 6`
`python3 -i -m otter trace/<trace name/<trace name>.otf2`

When using `OTF2_UNDEFINED_UINT64` in `trace_structs.c:179`:
```
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/adam/git/otter/src/python/otter/__main__.py", line 88, in <module>
    task_graph = tr.make_task_graph()
  File "/home/adam/git/otter/src/python/otter/trace.py", line 254, in make_task_graph
    return ig.Graph(**params)
  File "/home/adam/.local/lib/python3.8/site-packages/igraph/__init__.py", line 324, in __init__
    GraphBase.__init__(self, nverts, edges, directed)
ValueError: vertex IDs must be positive, got: -1

```
When using `OTF2_UNDEFINED_UINT32` in `trace_structs.c:179`:
```
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/adam/git/otter/src/python/otter/__main__.py", line 88, in <module>
    task_graph = tr.make_task_graph()
  File "/home/adam/git/otter/src/python/otter/trace.py", line 254, in make_task_graph
    return ig.Graph(**params)
  File "/home/adam/.local/lib/python3.8/site-packages/igraph/__init__.py", line 324, in __init__
    GraphBase.__init__(self, nverts, edges, directed)
OverflowError: long integer too large for conversion to C int
```
Ideas to fix:
- refactor c source to remove OTF2_UNDEFINED_UINT32/64 as task IDs
- have python replace these values with another placeholder

**Comment by adamtuft** [comment]
Another idea to fix:

- have the root initial task record itself as its own parent to remove the need for a dummy task ID.

**Comment by adamtuft** [comment]
It seems this is only an issue when trying to apply a task ID as a node ID in igraph. This is a bad idea generally anyway, igraph docs advise not to manually write node IDs.

**Comment by adamtuft** [comment]
Fixed in e4bace7
